draft/XXXX-configuration_handling/README.md

```diff
@@ -0,0 +1,201 @@
+---
+RFC: XXXX
+Author: Roni Choudhury
+Created: 2020-06-29
+Created: 2020-12-09
+Last-Modified: 2020-12-09
+
+# RFC XXXX: Configuration Handling
+
+One of the core features of our vision for Tinker Engine is the ability to
+accept a kind of "meta-configuration" from which concrete configurations are
+generated, used to drive individual protocol runs, with results aggregated
+together in some way that enables post-hoc data analysis. This RFC lays out some
+ideas for some of the core features such a handler will have, along with an
+overall architecture for configuration hanlding in Tinker Engine.
+
+## Proposals
+
+This RFC will discuss a few concrete proposals. One is the overall architcture
+by which the Tinker Engine runtime receives, processes, and submits
+configuration data to a protocol. Another is the method by which a configuration
+file can express variation in its parameters. A third is a fallback mechanism
+allowing programmatic configuration handling when the declarative methods are
+not sufficient for some use case.
+
+### Overall Architecture
+
+Currently, invoking the `tinker` CLI requires specifying a configuration file,
+but currently Tinker does nothing with this file. The new behavior will be for
+Tinker to load the specified file and parse it as YAML (bearing in mind that
+valid JSON is also valid YAML). If the file cannot be opened, or if the contents
```

Comment:
Yes. Legacy Tinker Engine passed the filename (like, the string value that was passed on the commandline) to the protocol, and it was up to the protocol to load the config file and do something with it. I consider this essentially doing nothing with it.

Reply:
It was the `(bearing in mind that valid JSON is also valid YAML)` being the part I was surprised about.  Not a big deal but I thought the colons would break YAML.  Which I could highlight sub lines.

Reply:
Ah gotcha. Here's the relevant section of the YAML spec: https://yaml.org/spec/1.2/spec.html#id2759572.

Note that technically, a JSON object with duplicate keys would be valid JSON but invalid YAML (a fact that I just learned). As the link says, in practice this means you have a bad JSON file. So for all practical purposes, JSON is a strict subset of YAML.